.travis.yml

```diff
@@ -5,50 +5,55 @@ addons:
   apt:
     packages:
     - socat
-    - enchant
-    - aspell
-    - aspell-en
 
 env:
   global:
-  - REDIS_TAGS="2.6.17 2.8.22 3.0.7 3.2.8 4.0.11 5.0.1" INSTALL_DIR=$HOME/redis
+  - REDIS_TAGS="2.6.17 2.8.22 3.0.7 3.2.8 4.0.11 5.0.1" INSTALL_DIR=$HOME/redis TEST_ARGS="-n 4"
 
 python:
 - "3.5"
 - "3.6"
 - "3.7"
 - "nightly"
+- "pypy3.5-6.0"
 
-matrix:
+stages:
+- lint
+- test
+
+jobs:
   include:
-  - python: "3.6"
-    env: DOCS_SPELL_CHECK="y"
+  - stage: lint
+    name: documentation spell check
+    python: "3.6"
+    addons:
+      apt:
+        packages:
+        - enchant
+        - aspell
+        - aspell-en
     install:
     - pip install -r docs/requirements.txt
     - pip install -e. -c tests/requirements.txt
+    script: make spelling
+  - &FLAKE
+    name: flake
+    python: "3.6"
+    install:
+    - pip install -r tests/requirements.txt
     script:
-    - make spelling
-  - python: "pypy3.5-5.8.0"
-    env: PYPY_RELEASE="pypy3-v6.0.0-linux64"
-  - python: "3.6"
+    - make flake
```

Comment:
spelling and flake could also be run in a single job.

Reply:
I don't want to mess all tasks into single build step as it was before.
It is much easier to understand what has failed from travis dashboard than digging into
job's log.

Reply:
Also, will try to move examples into separate stage

Reply:
Wow! At least someone supports me on non-messy jobs...

Reply:
@webknjaz 
I also like that better, but there is also quite some overhead sometimes that could be avoided.
It is much better with CircleCI in general, where you have slim/custom Docker base images, and not a bunch of requirements need to be installed.